src/clr-core/alert/alert.element.ts

```diff
@@ -0,0 +1,181 @@
+import { CwcButton } from '@clr/core/button';
+import {
+  baseStyles,
+  CommonStringsService,
+  event,
+  EventEmitter,
+  property,
+  querySlotAll,
+  registerElementSafely,
+} from '@clr/core/common';
+import {
+  checkCircleIcon,
+  ClarityIcons,
+  exclamationCircleIcon,
+  exclamationTriangleIcon,
+  infoCircleIcon,
+  timesIcon,
+} from '@clr/core/icon-shapes';
+import { html, LitElement } from 'lit-element';
+import { returnOrFallthrough } from '../common/utils/conditional';
+import { styles } from './alert.element.css';
+
+const iconMap = {
+  info: {
+    shape: ClarityIcons.getIconNameFromShape(infoCircleIcon),
+    title: CommonStringsService.keys.info,
+  },
+  success: {
+    shape: ClarityIcons.getIconNameFromShape(checkCircleIcon),
+    title: CommonStringsService.keys.success,
+  },
+  warning: {
+    shape: ClarityIcons.getIconNameFromShape(exclamationTriangleIcon),
+    title: CommonStringsService.keys.warning,
+  },
+  danger: {
+    shape: ClarityIcons.getIconNameFromShape(exclamationCircleIcon),
+    title: CommonStringsService.keys.danger,
+  },
+};
+
+/**
+ * Web component alerts.
+ *
+ * ```typescript
+ * import '@clr/core/alert';
+ * ```
+ *
+ * ```html
+ * <cwc-alert>
+ *    <cwc-alert-text>This is an alert.</cwc-alert-text>
+ * </cwc-alert>
+ * ```
+ *
+ * @noInheritDoc
+ * @beta 3.0
+ * @element cwc-alert
+ * @slot default - Content slot for inside the alert. Usually will contain at least a <cwc-alert-text> component.
+ * @cssprop --color
+ * @cssprop --background
+ * @cssprop --border-radius
+ * @cssprop --border-color
+ * @cssprop --icon-color
+ * @cssprop --close-icon-color
+ * @cssprop --close-icon-color-hover
+ */
+// @dynamic
+export class CwcAlert extends LitElement {
+  @event() private closedChange: EventEmitter<boolean>;
+
+  /** Sets the color of the alert from a predefined list of statuses */
+  @property({ type: String })
+  status: 'info' | 'success' | 'warning' | 'danger';
+
+  /** Sets the overall height and width of the alert and icon based on value */
+  @property({ type: String })
+  size: 'default' | 'sm';
+
+  /** Sets the icon shape to be used to override the default icon. The application must import the icon shape. */
+  @property({ type: String, attribute: 'icon-shape' })
+  iconShape = '';
+
+  /** Sets the title attribute for the icon. This may need to be set especially if the component is using custom icon shape.  */
+  @property({ type: String, attribute: 'icon-title' })
+  iconTitle = '';
+
+  /** If true, sets the style of the alert for app-level alerts */
+  @property({ type: Boolean })
+  appLevel = false;
+
+  /** If false, the alert will not render the close button.  */
+  @property({ type: Boolean })
+  closable = true;
+
+  @querySlotAll('cwc-button') private buttons: NodeListOf<CwcButton>;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    // buttons inside alert (usually app-level) have the style of small inverse buttons
+    this.buttons.forEach(button => {
```

Comment:
Will this work if buttons are dynamically hidden or shown with something like an ngIf?

Reply:
Good point. You can listen to the slot change event to set the buttons

```typescript

html`<slot name="actions" @slotchange="${this.updateButtons()}"></slot>`

updateButtons() {
  this.buttons.forEach(button => {
    button.status = 'inverse';
    button.size = 'sm';
  });
}
```